plugin/trino-faker/pom.xml

```diff
@@ -48,6 +48,12 @@
         <dependency>
             <groupId>io.trino</groupId>
             <artifactId>trino-main</artifactId>
+            <exclusions>
```

Comment:
I checked if other plugins also include main's transitive dependencies, and `vertica` does. It has `trino-main` as a runtime dependency, so I'm not sure anymore what's the correct way of doing this - should this be a runtime dependency here too? But why are transitive dependencies included for runtime dependencies?

Reply:
They should be provided scope

Reply:
> Trino plugin dependency io.trino:trino-main must not have scope 'provided'. It is not part of the SPI and will not be available at runtime.

Reply:
Runtime doesn't work, since it's needed at compile time. The dependency in Vertica is only used for tests, so I fixed that too.

Reply:
Fair enough .. so why is it needed at all then .. seems like a hack .. should it be scoped for tests only ? And whatever is needed .. should that go through the SPI?

Reply:
WDYM? Why is the exclusion needed? Because some types required to define connector functions are in the main package, and not the SPI package.

Reply:
Its more like .. why do we need the dependency .. its kinda bad to expose the classes without going through the SPI .. but that might be the current design and on purpose .. so the exclusion is probably good.